woocommerce/navigation
# Add wc-admin Links for internal routing

Fixes #91

This replaces https://github.com/woocommerce/navigation/pull/92 for a fresh start off of `main`.

Use `Link` from `@woocommerce/components` to render internal wc-admin links.

### No More Linking

This PR makes changes to webpack such that you can use the Gutenberg plugin to populate local dependencies, such as `window.wp.components` and consume them via a webpack `external`. I'm finding this much easier to deal with than linking.
**Note**: A master build of Gutenberg completely breaks wc-admin. We can go back to linking until this is fixed, but this PR can be tested as is.

### Still to do

I'm asking for review early so we can discuss the pros/cons of this approach, but there are some items missing:

* persistedQueries
* excludedScreens

### Test instructions

1. You'll need the `fix/navigation-package-deps` branch from https://github.com/woocommerce/woocommerce-admin/pull/5169 so dependencies load correctly.
2. Install Gutenberg plugin `master` branch and `npm run build` it.
3. Use the Navigation and see the internal links work when navigating from wc-admin page to wc-admin page.
